jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/S3VirtualBucketTest.kt

```diff
@@ -240,11 +247,39 @@ class S3VirtualBucketTest {
 
         // Use real manager for this since it can affect the S3Configuration that goes into S3Utilities
         AwsClientManager().getClient<S3Client>(awsConnectionManager.activeCredentialProvider, awsConnectionManager.activeRegion).use {
-            val sut = S3VirtualBucket(Bucket.builder().name("test-bucket").build(), it)
+            val sut = S3VirtualBucket(Bucket.builder().name("test-bucket").build(), it, projectRule.project)
 
             assertThatThrownBy {
                 sut.generateUrl("", null)
             }
         }
     }
+
+    @Test
+    fun closeWindowOnNoSuchBucketException() {
+        val s3Mock = mockClientManager.create<S3Client>()
+        val testBucket = Bucket.builder().name("TestBucket").build()
+        val s3VirtualBucket = S3VirtualBucket(testBucket, s3Mock, projectRule.project)
+        runInEdtAndWait {
+            runWriteAction {
+                FileTypeManagerEx.getInstanceEx().associatePattern(
+                    FileTypes.PLAIN_TEXT,
+                    testBucket.name()
+                )
+            }
+            assertThat(openEditor(projectRule.project, testBucket)).isNotNull
+        }
+        val fileEditorManager = FileEditorManager.getInstance(projectRule.project)
+        assertThat(fileEditorManager.openFiles).contains(s3VirtualBucket)
+
+        try {
```

Comment:
we don't need to actually throw, we can just run `s3VirtualBucket.handleDeletedBucket()`